protocol/contracts/templegold/DaiGoldAuction.sol

```diff
@@ -0,0 +1,339 @@
+     */
+    function startAuction() external override {
+        if (auctionStarter != address(0) && msg.sender != auctionStarter) { revert CommonEventsAndErrors.InvalidAccess(); }
+        EpochInfo storage prevAuctionInfo = epochs[_currentEpochId];
+        if (!prevAuctionInfo.hasEnded()) { revert CannotStartAuction(); }
+       
+        AuctionConfig storage config = auctionConfig;
+        /// @notice last auction end time plus wait period
+        if (_currentEpochId > 0 && (prevAuctionInfo.endTime + config.auctionsTimeDiff > block.timestamp)) {
+            revert CannotStartAuction();
+        }
+        _distributeGold();
+        uint256 totalGoldAmount = nextAuctionGoldAmount;
+        nextAuctionGoldAmount = 0;
+        uint256 epochId = _currentEpochId = _currentEpochId + 1;
+        
+        if (totalGoldAmount < config.auctionMinimumDistributedGold) { revert LowGoldDistributed(totalGoldAmount); }
+
+        EpochInfo storage info = epochs[epochId];
+        info.totalAuctionTokenAmount = totalGoldAmount;
+        uint128 startTime = info.startTime = uint128(block.timestamp) + config.auctionStartCooldown;
+        uint128 endTime = info.endTime = startTime + AUCTION_DURATION;
+
+        emit AuctionStarted(epochId, msg.sender, startTime, endTime, totalGoldAmount);
+    }
+
+    /**
+     * @notice Deposit bidding token for current running epoch auction
+     * @param amount Amount of bid token to deposit
+     */
+    function bid(uint256 amount) external virtual override onlyWhenLive {
+        if (amount == 0) { revert CommonEventsAndErrors.ExpectedNonZero(); }
+
+        bidToken.safeTransferFrom(msg.sender, treasury, amount);
+
+        uint256 epochIdCache = _currentEpochId;
+        depositors[msg.sender][epochIdCache] += amount;
+
+        EpochInfo storage info = epochs[epochIdCache];
+        info.totalBidTokenAmount += amount;
+        emit Deposit(msg.sender, epochIdCache, amount);
+    }
+
+    /**
+     * @notice Claim share of Temple Gold for epoch
+     * Can only claim for past epochs, not current auction epoch.
+     * @param epochId Id of epoch
+     */
+    function claim(uint256 epochId) external virtual override {
+        /// @notice cannot claim for current live epoch
+        EpochInfo storage info = epochs[epochId];
+        if (!info.hasEnded()) { revert CannotClaim(epochId); }
+        /// @dev epochId could be invalid. eg epochId > _currentEpochId
+        if (info.startTime == 0) { revert InvalidEpoch(); }
+        if (claimed[msg.sender][epochId]) { revert AlreadyClaimed(); }
+
+        uint256 bidTokenAmount = depositors[msg.sender][epochId];
+        if (bidTokenAmount == 0) { revert CommonEventsAndErrors.ExpectedNonZero(); }
+        claimed[msg.sender][epochId] = true;
+        uint256 claimAmount = bidTokenAmount.mulDivRound(info.totalAuctionTokenAmount, info.totalBidTokenAmount, false);
+        claimedAmount[msg.sender][epochId] = claimAmount;
```

Comment:
nit:

In the spice auctions, there is a mapping tracking the total claimed amounts per token that is updated in the equivalent `claim()` function:

```solidity
        claimedAmount[msg.sender][epochId] = claimAmount;
        /// checkpoint claim for auction token
        _claimedAuctionTokens[auctionToken] += claimAmount;
```

For whatever reason you're tracking `_claimedAuctionTokens` in the spice auctions, perhaps you also want to track the total claimed TGLD in DaiGoldAuction contract as well.

Reply:
That is because, in the spice auctions, the auction token could be TGLD in rare cases but in dai gold TGLD is the only auction token. 
Check configuration `config.isTempleGoldAuctionToken`

Reply:
sorry, I was not clear. I was not referring to have a mapping. I meant that you track the total claimed tokens in a state variable: 

```solidity
uint256 totalClaimedGold;
```
And that you update that variable in `claim()`

```solidity
totalClaimedGold += claimAmount;
```
But only if it is relevat to track the total Claimed in the frontend or something